tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -332,9 +329,52 @@ private void AsmDataProductConfigChanged(object sender, ProductConfigChangedEven
             }
         }
 
+        private void AsmProductConfigChanged(object sender, ProductConfigChangedEventArgs e)
+        {
+            if (!_enabled) { return; }
+
+            _ruleStatus ??= new Dictionary<string, bool>(StringComparer.InvariantCultureIgnoreCase);
```

Comment:
I don't think the use of `_ruleStatus` is threadSafe? Don't we need a lock or a concurrent dictionary here 🤔 and in `UpdateRulesData`? They can be called concurrently AFAICT

Reply:
I was asuming Remote Config events can't occur simultaneously. But it is a personal assumption. A lock wouldn't hurt.